firefox-ios/Client/TabManagement/Tab.swift

```diff
@@ -947,29 +957,51 @@ class Tab: NSObject, ThemeApplicable, FeatureFlaggable, ShareTab {
     }
 
     private func deleteDownloadedDocuments() {
-        let docsURL = downloadedTemporaryDocs
+        let docsURL = temporaryDocumentsSession
         guard !docsURL.isEmpty else { return }
-        DispatchQueue.global(qos: .background).async {
+        DispatchQueue.global(qos: .background).async { [fileManager] in
```

Comment:
Just an idea, could we test `deleteDownloadedDocuments` by injecting `DispatchQueue.global(qos: .background)` with a `DispatchQueueInterface`? Then on `deinit` of the tab object we could check `removeItem` is called?